### PR TITLE
:sparkles: real music

### PR DESCRIPTION
**Pull request by guillaumecrespel**
Add library to play real music => https://github.com/keithwhor/audiosynth

rythmbox Playing by character and not by instrument, because, the instrument changed by the character.

### CODE REVIEW

File: public/audiosynth.js [hunk 1]
```diff
@@ -0,0 +1,350 @@
+var Synth, AudioSynth, AudioSynthInstrument;
```
Comment (fabienjuif) [inline public/audiosynth.js]: Do you search for this lib on npm ?

File: src/components/rythmbox/rythmbox.js [hunk 2]
```diff
@@ -3,25 +3,25 @@ import isolate from '@cycle/isolate'
 import xs from 'xstream'
 import Range from './range'
 
-const frequencies = [
-  329, // E
-  392, // G
-  587, // D
+const notes = [
+  'C', // E
```
Comment (fabienjuif) [inline src/components/rythmbox/rythmbox.js]: Delete commentaries

File: src/components/rythmbox/rythmbox.js [hunk 3]
```diff
@@ -3,25 +3,25 @@ import isolate from '@cycle/isolate'
...
+  'D', // G
+  'E', // D
 ]
-const instruments = [
-  'guitare',
-  'bass',
-  'ocarina',
-  'harp',
+const characters = [
```
Comment (fabienjuif) [inline src/components/rythmbox/rythmbox.js]: We should pass this from props to avoid duplication

File: src/drivers/music.js [hunk 4]
```diff
@@ -1,16 +1,10 @@
-const context = new (window.AudioContext || window.webkitAudioContext)()
-
 export default (sink$) => {
   sink$.addListener({
-    next: (note) => {
-      const { type = 'sine', frequency, time = 200 } = note
+    next: (music) => {
+      const { instrument, note, time = 2 } = music
```
Comment (fabienjuif) [inline src/drivers/music.js]: We should delete default value as we argue before

**Review by fabienjuif — CHANGES_REQUESTED** [review]
Nice, some minors commentaries

File: public/audiosynth.js (hunk 1 above)
Comment (guillaumecrespel) [reply public/audiosynth.js]: this lib is not present on npm, and i don't find equivalent

Comment (fabienjuif) [reply public/audiosynth.js]: We should fork it (https://github.com/keithwhor/audiosynth) and publish it.
So we can import it trough webpack

Comment (guillaumecrespel) [reply public/audiosynth.js]: Just for that ? it's one file, and i think, we don't maintain this lib ?

File: src/components/speaker/speaker.js [hunk 5]
```diff
@@ -7,14 +7,14 @@ export default ({ MUSIC$ }) => {
...
-    .map(music => xs.of(music).compose(delay(music.time)))
+    .map(music => xs.of(music).compose(delay(music.time * 1000)))
     .flatten()
     .map(music => Object.assign({}, music, { stop: true }))
 
   const music$ = xs.merge(musicStart$, musicStop$)
 
   const vdom$ = music$
-    .startWith({ stop: true })
+    .startWith({ stop: true }).debug()
```
Comment (fabienjuif) [inline src/components/speaker/speaker.js]: debug

File: src/components/rythmbox/rythmbox.js [hunk 6]
```diff
@@ -3,40 +3,28 @@ import isolate from '@cycle/isolate'
...
-    .map(instrument => ({ instrument, frequencies }))
-    .map(props =>
-      isolate(
-        Range,
-        props.instrument,
-      )({ DOM$, props$: xs.of(props) }),
-    )
+export default ({ DOM$, props$ }) => {
+  const ranges$ = props$
+    .map(characters => characters.map(c => isolate(Range, c.name)({
```
Comment (fabienjuif) [inline src/components/rythmbox/rythmbox.js]: I prefer to map in two instructions, I find it easier to read and explain

File: src/drivers/music.js [hunk 7]
```diff
@@ -1,16 +1,12 @@
...
+    next: (music) => {
+      const { instrument, note, time } = music
 
-      const osc = context.createOscillator()
-      osc.type = type
-      osc.frequency.value = frequency
-      osc.connect(context.destination)
-      osc.start()
-      osc.stop(context.currentTime + (time / 1000))
+      // eslint-disable-next-line no-undef
```
Comment (fabienjuif) [inline src/drivers/music.js]: We cand delete the eslint-disable, I forgot it

File: src/components/rythmbox/rythmbox.js [hunk 8]
```diff
@@ -3,40 +3,32 @@ import isolate from '@cycle/isolate'
...
-  392, // G
-  587, // D
-]
-const instruments = [
-  'guitare',
-  'bass',
-  'ocarina',
-  'harp',
-]
+const notes = ['A', 'B', 'C', 'D', 'E', 'F', 'G']
```
Comment (fabienjuif) [inline src/components/rythmbox/rythmbox.js]: Merge the other PR then move this into the config file ?

**Review by fabienjuif — APPROVED** [review]

**Pull request merged**